components/chassis.py

```diff
@@ -130,3 +133,11 @@ def reset_odometry(self, pose: Pose2d) -> None:
         self.left_encoder.setPosition(0)
         self.right_encoder.setPosition(0)
         self.odometry.resetPosition(pose, self._get_heading())
+
+    def angle_to_target(self) -> float:
+        """Return the theoretical angle (in radians) to the target based on odometry"""
+        pose: Pose2d = self.chassis.get_pose()
+        rel: Transform2d = pose - self.TARGET_POSITION
+        translation = rel.translation()
+        rel_heading = Rotation2d(translation.y, translation.x) + pose.rotation()
+        return rel_heading.radians()
```

Comment:
Now that this has been factored out, I can see what this is actually trying to do… and I'm not sure that the maths is correct.

I think it would be prudent to have a method to find the heading to _any_ position (as a Translation2d) on the field. You would then just subtract the two translations and turn that into a Rotation2d.

We could then have a convenience method `get_angle_to_power_port` (let's use the game's terms) which calls `get_angle_to` or whatever we call it with the correct coordinates.